File: BeginnersSelection/abs/abc086c/Main.rb
def solve(positions)
  ct, cx, cy = [0, 0, 0]
  positions.each do |position|
    t, x, y = position
    diff = (cx - x).abs + (cy - y).abs
    diff_t = t - ct
    if diff > diff_t
      return "No"
    elsif diff - diff_t == 0 || (diff - diff_t) % 2 == 0
      ct, cx, cy = position
    else
      return "No"
    end
  end
  return "Yes"
end

n = gets.strip.to_i
positions = []
n.times do
  positions.push(gets.strip.split.map(&:to_i))
end

puts solve(positions)

File: BeginnersSelection/abs/abc086c/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	fmt.Scan(&n)
	ct, cx, cy := 0.0, 0.0, 0.0

	for i := 0; i < n; i++ {
		var t, x, y float64
		fmt.Scan(&t, &x, &y)

		dt := t - ct
		dx := math.Abs(x - cx)
		dy := math.Abs(y - cy)

		if dx+dy > dt {
			fmt.Println("No")
			return
		} else if (int(dx)+int(dy)-int(dt))%2 != 0 {
			fmt.Println("No")
			return
		}
		ct, cx, cy = t, x, y
	}
	fmt.Println("Yes")
}
